### PR TITLE
Unblock after pristine deconfiguration

### PR DESCRIPTION
**Pull request by sabaini**
When a charm has a non-pristine disk deconfigured, it should leave the previously-set blocked state (LP #1988088).

### CODE REVIEW

File: zaza/openstack/charm_tests/ceph/tests.py
Comment (ChrisMacNaughton) [inline zaza/openstack/charm_tests/ceph/tests.py:428]: It feels like this set_default should look at whatever's currently set in the model rather than just using charm defaults

Comment (sabaini) [reply zaza/openstack/charm_tests/ceph/tests.py:429]: I've now added two calls to store app config away and re-store it at the end of the test method so we're going back to where we started -- does this address your concerns?

File: zaza/openstack/charm_tests/ceph/tests.py
Comment (ChrisMacNaughton) [inline zaza/openstack/charm_tests/ceph/tests.py:435]: Does this not need to call `hooks/update-status`?

**Review by ChrisMacNaughton — APPROVED** [review]
LGTM - holding off on landing just until the review lands as well

**Pull request merged**
